language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -24,47 +24,59 @@ import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutorService, Future}
 
 private final class CodeGenRunner(
-    scope: CodeGenRunner.Scope,
+    scopeByPackagePrefix: Map[Option[String], CodeGenRunner.Scope],
     outputDirectory: Path,
     decoderPackageAndClass: Option[(String, String)],
 ) extends StrictLogging {
 
   def runWith(executionContext: ExecutionContext): Future[Unit] = {
     implicit val ec: ExecutionContext = executionContext
-    val packageIds = scope.signatures.map(_.packageId).mkString(", ")
-    logger.info(s"Start processing packageIds '$packageIds'")
-    for {
-      _ <- generateDecoder()
-      interfaceTrees = scope.signatures.map(InterfaceTree.fromInterface)
-      _ <- Future.traverse(interfaceTrees)(processInterfaceTree(_))
-    } yield logger.info(s"Finished processing packageIds '$packageIds'")
+    Future
+      .traverse(scopeByPackagePrefix.toSeq) { case (maybePrefix, scope) =>
```

Comment:
instead of generating code with mixed package prefix. it is now generating code for each entries in the `scopeByPackagePrefix` per prefix `Optiona[String]`